typescript/ts-node-lowering/src/org/jetrbains/dukat/nodeLowering/lowerings/convertStatements.kt

```diff
@@ -502,6 +577,46 @@ private class ExpressionConverter() {
             )
         }
     }
+
+    private fun convertTypeOf(expression: ExpressionModel, type: String): ExpressionModel {
+        return IsExpressionModel(
+            expression = expression,
+            type = TypeValueModel(
+                IdentifierEntity(
+                    when (type) {
+                        "boolean" -> "Boolean"
+                        "number" -> "Number"
+                        "string" -> "String"
+                        else -> raiseConcern("unsupported type in typeof: $type") {
+                            "UNSUPPORTED_TYPE"
+                        }
+                    }),
+                listOf(),
+                null,
+                null
+            )
+        )
+    }
+
+    private fun BinaryExpressionDeclaration.convert(): ExpressionModel {
+        if (convertBinaryOperator(operator) == EQ) {
+            val left = left
```

Comment:
why we need this variables?

Reply:
to enable smart-casting in type checks below - is there more elegant way to do this?